src/main/java/uk/gov/hmcts/reform/preapi/media/MediaKind.java

```diff
@@ -58,8 +62,62 @@ public MediaKind(
     }
 
     @Override
-    public String playAsset(String assetId) {
-        throw new UnsupportedOperationException();
+    public PlaybackDTO playAsset(String assetName, String userId) {
+        if (getAsset(assetName) == null) {
+            throw new NotFoundException("Asset: " + assetName);
+        }
+
+        var locators = mediaKindClient.getAssetStreamingLocators(assetName);
+
+        var locator = locators.getStreamingLocators()
+            .stream()
+            .filter(l -> l.getName().equals(userId))
+            .findFirst()
+            .orElse(null);
+
+        if (locator == null) {
+            var properties = MkCreateStreamingLocator.MkCreateStreamingLocatorProperties.builder()
+                .assetName(assetName)
+                .streamingPolicyName("Predefined_ClearStreamingOnly")
+                .endTime(Instant.now().plusSeconds(3600).toString())
+                .build();
+            try {
+                mediaKindClient.putStreamingLocator(userId, MkCreateStreamingLocator.builder()
+                    .properties(properties).build());
+            } catch (FeignException e) {
+                throw new MediaKindException();
+            }
```

Comment:
This code is hiding what is happening in the FeignException. We won't get to know if it's a 500,400,404 etc. In this case, as the business logic doesn't depend on what the exception is, you can let the `FeignException` bubble up as there is an entry in the `GlobalExceptionHandler` for it.
```suggestion
            mediaKindClient.putStreamingLocator(userId, MkCreateStreamingLocator.builder()
                .properties(properties).build());
```